We will be using House Pricing dataset.

In [1]:
import pandas as pd

In [2]:
House_data = pd.read_csv("Housing.csv")
House_data

price  area  bedrooms  bathrooms  stories mainroad guestroom basement  \
0    13300000  7420         4          2        3      yes        no       no   
1    12250000  8960         4          4        4      yes        no       no   
2    12250000  9960         3          2        2      yes        no      yes   
3    12215000  7500         4          2        2      yes        no      yes   
4    11410000  7420         4          1        2      yes       yes      yes   
..        ...   ...       ...        ...      ...      ...       ...      ...   
540   1820000  3000         2          1        1      yes        no      yes   
541   1767150  2400         3          1        1       no        no       no   
542   1750000  3620         2          1        1      yes        no       no   
543   1750000  2910         3          1        1       no        no       no   
544   1750000  3850         3          1        2      yes        no       no   

    hotwaterheating airconditioning  parking prefarea furnishingstatus  
0                no             yes        2      yes        furnished  
1                no             yes        3       no        furnished  
2                no              no        2      yes   semi-furnished  
3                no             yes        3      yes        furnished  
4                no             yes        2       no        furnished  
..              ...             ...      ...      ...              ...  
540              no              no        2       no      unfurnished  
541              no              no        0       no   semi-furnished  
542              no              no        0       no      unfurnished  
543              no              no        0       no        furnished  
544              no              no        0       no      unfurnished  

[545 rows x 13 columns]

In [3]:
House_data.shape

(545, 13)

In [4]:
House_data.head(5)

price  area  bedrooms  bathrooms  stories mainroad guestroom basement  \
0  13300000  7420         4          2        3      yes        no       no   
1  12250000  8960         4          4        4      yes        no       no   
2  12250000  9960         3          2        2      yes        no      yes   
3  12215000  7500         4          2        2      yes        no      yes   
4  11410000  7420         4          1        2      yes       yes      yes   

  hotwaterheating airconditioning  parking prefarea furnishingstatus  
0              no             yes        2      yes        furnished  
1              no             yes        3       no        furnished  
2              no              no        2      yes   semi-furnished  
3              no             yes        3      yes        furnished  
4              no             yes        2       no        furnished

In [5]:
House_data.tail(5)

price  area  bedrooms  bathrooms  stories mainroad guestroom basement  \
540  1820000  3000         2          1        1      yes        no      yes   
541  1767150  2400         3          1        1       no        no       no   
542  1750000  3620         2          1        1      yes        no       no   
543  1750000  2910         3          1        1       no        no       no   
544  1750000  3850         3          1        2      yes        no       no   

    hotwaterheating airconditioning  parking prefarea furnishingstatus  
540              no              no        2       no      unfurnished  
541              no              no        0       no   semi-furnished  
542              no              no        0       no      unfurnished  
543              no              no        0       no        furnished  
544              no              no        0       no      unfurnished

In [6]:
House_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 545 entries, 0 to 544
Data columns (total 13 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   price             545 non-null    int64 
 1   area              545 non-null    int64 
 2   bedrooms          545 non-null    int64 
 3   bathrooms         545 non-null    int64 
 4   stories           545 non-null    int64 
 5   mainroad          545 non-null    object
 6   guestroom         545 non-null    object
 7   basement          545 non-null    object
 8   hotwaterheating   545 non-null    object
 9   airconditioning   545 non-null    object
 10  parking           545 non-null    int64 
 11  prefarea          545 non-null    object
 12  furnishingstatus  545 non-null    object
dtypes: int64(6), object(7)
memory usage: 55.5+ KB


In [7]:
Mainroad_price = House_data[["price","mainroad"]]
Mainroad_price

price mainroad
0    13300000      yes
1    12250000      yes
2    12250000      yes
3    12215000      yes
4    11410000      yes
..        ...      ...
540   1820000      yes
541   1767150       no
542   1750000      yes
543   1750000       no
544   1750000      yes

[545 rows x 2 columns]

In [8]:
House_data["furnishingstatus"]


0           furnished
1           furnished
2      semi-furnished
3           furnished
4           furnished
            ...      
540       unfurnished
541    semi-furnished
542       unfurnished
543         furnished
544       unfurnished
Name: furnishingstatus, Length: 545, dtype: object

In [9]:
Mainroad_price.shape

(545, 2)

```iloc``` in pandas

In [15]:
row1 = House_data.iloc[[4, 5, 6, 7]]

row2 = House_data.iloc[4:8]

row1==row2
 

price  area  bedrooms  bathrooms  stories  mainroad  guestroom  basement  \
4   True  True      True       True     True      True       True      True   
5   True  True      True       True     True      True       True      True   
6   True  True      True       True     True      True       True      True   
7   True  True      True       True     True      True       True      True   

   hotwaterheating  airconditioning  parking  prefarea  furnishingstatus  
4             True             True     True      True              True  
5             True             True     True      True              True  
6             True             True     True      True              True  
7             True             True     True      True              True

```concat``` it  does concatenation of two datasets

Let's take an example of mainroad_prices and house_data here we can see that no. of row both have is 545 threfore on adding this two will get us 1090 rows

In [17]:
pd.concat([House_data,Mainroad_price]).shape



(1090, 13)

But if we had different named series in both the datasets the we can append these columns by ```concat([dataset1,dataset2],axis=1)```.
Since in above example value of axis was assigned as 0.
So no. of index 13 + 2 = 15

In [23]:
pd.concat([House_data,Mainroad_price],axis=1).head(5)

price  area  bedrooms  bathrooms  stories mainroad guestroom basement  \
0  13300000  7420         4          2        3      yes        no       no   
1  12250000  8960         4          4        4      yes        no       no   
2  12250000  9960         3          2        2      yes        no      yes   
3  12215000  7500         4          2        2      yes        no      yes   
4  11410000  7420         4          1        2      yes       yes      yes   

  hotwaterheating airconditioning  parking prefarea furnishingstatus  \
0              no             yes        2      yes        furnished   
1              no             yes        3       no        furnished   
2              no              no        2      yes   semi-furnished   
3              no             yes        3      yes        furnished   
4              no             yes        2       no        furnished   

      price mainroad  
0  13300000      yes  
1  12250000      yes  
2  12250000      yes  
3  12215000      yes  
4  11410000      yes

```sort_values()``` will Order rows by values of a column (low to high)

In [28]:
# Mainroad_price.sort_values('price')
Mainroad_price.sort_values('price', ascending=False).head(5)

price mainroad
0  13300000      yes
2  12250000      yes
1  12250000      yes
3  12215000      yes
4  11410000      yes

***Rename***


In [29]:

Mainroad_price.rename(columns= {"mainroad":"onmainroad","price":"priceofhouse"})

priceofhouse onmainroad
0        13300000        yes
1        12250000        yes
2        12250000        yes
3        12215000        yes
4        11410000        yes
..            ...        ...
540       1820000        yes
541       1767150         no
542       1750000        yes
543       1750000         no
544       1750000        yes

[545 rows x 2 columns]

***Sorting and then resetting the index***

In [31]:
Mainroad_price.sort_index()
# Here it just sort the index

price mainroad
0    13300000      yes
1    12250000      yes
2    12250000      yes
3    12215000      yes
4    11410000      yes
..        ...      ...
540   1820000      yes
541   1767150       no
542   1750000      yes
543   1750000       no
544   1750000      yes

[545 rows x 2 columns]

In [32]:
Mainroad_price.reset_index()

index     price mainroad
0        0  13300000      yes
1        1  12250000      yes
2        2  12250000      yes
3        3  12215000      yes
4        4  11410000      yes
..     ...       ...      ...
540    540   1820000      yes
541    541   1767150       no
542    542   1750000      yes
543    543   1750000       no
544    544   1750000      yes

[545 rows x 3 columns]

***Summerize Data ***

In [38]:
# For each unique value in that column it count NO. or rows
Mainroad_price['price'].value_counts().head(5)

price
3500000    17
4200000    17
4900000    12
3150000     9
5600000     9
Name: count, dtype: int64

In [37]:
# Number  distinct values in a column.
Mainroad_price['price'].nunique()

219

In [49]:
# min value in that series or column 
Mainroad_price["price"].min()

# max value in that series or column
Mainroad_price["price"].max()

# average value in that series or column
Mainroad_price["price"].mean()

# median value in that series or column
Mainroad_price["price"].median()

# Standard deviation of each object or value 
Mainroad_price["price"].std()

# Varience of each object or value 
Mainroad_price["price"].var()

# Sum of each object or value 
Mainroad_price["price"].sum()

# It counts non null value
Mainroad_price["price"].count()

# Quantile of each object or value
# A quantile is a value that divides a probability distribution 
# or sorted data into equal parts.
Mainroad_price["price"].quantile() 

# Apply function to each object.
def double(integer):
    return 2*integer

Mainroad_price.apply(double)

price mainroad
0    26600000   yesyes
1    24500000   yesyes
2    24500000   yesyes
3    24430000   yesyes
4    22820000   yesyes
..        ...      ...
540   3640000   yesyes
541   3534300     nono
542   3500000   yesyes
543   3500000     nono
544   3500000   yesyes

[545 rows x 2 columns]